docs/cloud/dev/spryker-cloud-commerce-os/configuring-deployment-pipelines/deployment-in-states.md

```diff
@@ -0,0 +1,132 @@
+
+An important part of this deployment step is this command:
+```shell
+vendor/bin/console scheduler:suspend -vvv --no-ansi
+```
+as it will stop our scheduler from running. Hereby preventing any data corruption or error for the duration of the deployment.
+It will first wait for the currently running jobs to finish and gracefully shutdown. (add this to the duration of the deployment)
+
+![scheduler:suspend](./images/Run_pre-deploy_hook/Run_pre-deploy_hook.jpg)
+
+Requests in the meantime are still coming in:
+
+![Run_pre-deploy_hook](./images/Run_pre-deploy_hook/pre_deploy.gif)
+
+### Deploy_Scheduler
+
+Now it's time to deploy the scheduler. It is based on the Zed container, as it needs to use Zed codebase. 
+
+![scheduler_paused](./images/Deploy_Scheduler/Deploy_Scheduler.jpg)
+
+Because the scheduler was paused, we don't need to fear it will run against incorrect data or services. 
+However, requests still are coming in:
+
+![Deploy_Scheduler](./images/Deploy_Scheduler/deploy_scheduler.gif)
+
+### Run_install
+
+This step should take care of running the necessary migrations for our new version (V2 here) of our application.
+The script that we run is associated with the constant: `SPRYKER_HOOK_INSTALL`.
+By default, this will run:
+```shell
+vendor/bin/install -r EU/production --no-ansi -vvv
+```
+
+After this step, the database should be updated. However, Search and Redis are not, as we "paused" the synchronization.
+
+![migrations](./images/Run_install/migrations.jpg)
+
+we should realize by now, that requests still coming in might error out. It all depends on what was migrated.
+
+At the very end of this step, we re-enable our scheduler, and setup the new jobs (if there was any)
+```shell
+vendor/bin/console scheduler:setup -vvv --no-ansi
+```
+When this command run, it will restart our queue workers, and update search and redis.
+
+![Run_install](./images/Run_install/install_dbs_updates/install_dbs_updates.gif)
+
+This process might take a while, depending on the amount of data that needs to be processed. And request are still coming in:
+
+![Run_install_requests](./images/Run_install/request_during_install/install_request.gif)
+
+### Deploy_Spryker_services
+Finally, we are going to deploy our services with the updated codebase. (Zed V2, Glue V2).
+For the sake of simplicity, let's assume that Redis and search are done updating. (We leave the asterisks on the schema, as a reminder that it might not be the case, it depends on the size of the migration)
+
+![Deploy_Spryker_services](./images/Deploy_Spryker_services/Deploy_Spryker_services.jpg)
+
+This process takes some time. AWS first spawn our v2 services, and when they are up and running, they take V1 down. 
+We do not control this process, so there is potentially a timeframe here where the application will run V1 and V2 or services in a random combination. (pessimistic approach)
```

Comment:
Do you mean that the application can randomly use incompatible versions of the services?

Reply:
correct